pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -191,26 +190,25 @@ func (r *DNSProvider) changeRecord(action, fqdn, value string, ttl int) error {
 	recordSet := newTXTRecordSet(fqdn, value, ttl)
 	reqParams := &route53.ChangeResourceRecordSetsInput{
 		HostedZoneId: aws.String(hostedZoneID),
-		ChangeBatch: &route53.ChangeBatch{
+		ChangeBatch: &route53types.ChangeBatch{
 			Comment: aws.String("Managed by cert-manager"),
-			Changes: []*route53.Change{
+			Changes: []route53types.Change{
 				{
-					Action:            &action,
+					Action:            action,
 					ResourceRecordSet: recordSet,
 				},
 			},
 		},
 	}
 
-	resp, err := r.client.ChangeResourceRecordSets(reqParams)
+	resp, err := r.client.ChangeResourceRecordSets(context.TODO(), reqParams)
 	if err != nil {
-		if awserr, ok := err.(awserr.Error); ok {
-			if action == route53.ChangeActionDelete && awserr.Code() == route53.ErrCodeInvalidChangeBatch {
-				r.log.V(logf.DebugLevel).WithValues("error", err).Info("ignoring InvalidChangeBatch error")
-				// If we try to delete something and get a 'InvalidChangeBatch' that
-				// means it's already deleted, no need to consider it an error.
-				return nil
-			}
+		invalidChangeBatchErr := &route53types.InvalidChangeBatch{}
+		if errors.As(err, &invalidChangeBatchErr) && action == route53types.ChangeActionDelete {
+			r.log.V(logf.DebugLevel).WithValues("error", err).Info("ignoring InvalidChangeBatch error")
+			// If we try to delete something and get a 'InvalidChangeBatch' that
+			// means it's already deleted, no need to consider it an error.
+			return nil
 		}
```

Comment:
nit: `invalidChangeBatchErr` seems to be unused. Consider using `errors.Is` instead.

Reply:
Updated.